src/dc_msg.rs

```diff
@@ -831,98 +832,74 @@ pub unsafe fn dc_msg_get_summarytext(
         return dc_strdup(0 as *const libc::c_char);
     }
 
-    let msgtext_c = (*msg)
-        .text
-        .as_ref()
-        .map(|s| CString::yolo(String::as_str(s)));
-    let msgtext_ptr = msgtext_c.map_or(ptr::null(), |s| s.as_ptr());
-
     dc_msg_get_summarytext_by_raw(
         (*msg).type_0,
-        msgtext_ptr,
+        (*msg).text.as_ref().unwrap(),
         &mut (*msg).param,
         approx_characters,
         (*msg).context,
     )
+    .strdup()
 }
 
-/* the returned value must be free()'d */
+/// get a summary text
 #[allow(non_snake_case)]
-pub unsafe fn dc_msg_get_summarytext_by_raw(
+pub fn dc_msg_get_summarytext_by_raw(
     type_0: Viewtype,
-    text: *const libc::c_char,
+    text: &str,
     param: &mut Params,
     approx_characters: libc::c_int,
```

Comment:
Why not change this to `u32`?  If you also change the dc_truncate* funcs to use u32 (see comment there) than none of the weird `.try_into().unwrap()` stuff would be needed.